include/bout/fv_ops.hxx

```diff
@@ -383,63 +398,60 @@ const Field3D Div_par(const Field3D& f_in, const Field3D& v_in,
    *
    */
 template <typename CellEdges = MC>
-const Field3D Div_f_v(const Field3D& n_in, const Vector3D& v, bool bndry_flux) {
+Field3D Div_f_v(const Field3D& n_in, const Vector3D& v, bool bndry_flux) {
   ASSERT1(n_in.getLocation() == v.getLocation());
   ASSERT1_FIELDS_COMPATIBLE(n_in, v.x);
 
-  Mesh* mesh = n_in.getMesh();
+  const Mesh* mesh = n_in.getMesh();
 
   CellEdges cellboundary;
 
-  Coordinates* coord = n_in.getCoordinates();
+  const Coordinates* coord = n_in.getCoordinates();
 
   if (v.covariant) {
     // Got a covariant vector instead
     throw BoutException("Div_f_v passed a covariant v");
   }
 
   Field3D result{zeroFrom(n_in)};
-
-  Field3D vx = v.x;
-  Field3D vz = v.z;
-  Field3D n = n_in;
+  const auto& J = coord->J;
 
   BOUT_FOR(i, result.getRegion("RGN_NOBNDRY")) {
     // Calculate velocities
-    BoutReal vU = 0.25 * (vz[i.zp()] + vz[i]) * (coord->J[i.zp()] + coord->J[i]);
-    BoutReal vD = 0.25 * (vz[i.zm()] + vz[i]) * (coord->J[i.zm()] + coord->J[i]);
-    BoutReal vL = 0.25 * (vx[i.xm()] + vx[i]) * (coord->J[i.xm()] + coord->J[i]);
-    BoutReal vR = 0.25 * (vx[i.xp()] + vx[i]) * (coord->J[i.xp()] + coord->J[i]);
+    const BoutReal v_U = 0.25 * (v.z[i.zp()] + v.z[i]) * (J[i.zp()] + J[i]);
+    const BoutReal v_D = 0.25 * (v.z[i.zm()] + v.z[i]) * (J[i.zm()] + J[i]);
+    const BoutReal v_L = 0.25 * (v.x[i.xm()] + v.x[i]) * (J[i.xm()] + J[i]);
+    const BoutReal v_R = 0.25 * (v.x[i.xp()] + v.x[i]) * (J[i.xp()] + J[i]);
 
     // X direction
-    Stencil1D s;
-    s.c = n[i];
-    s.m = n[i.xm()];
-    s.mm = n[i.xmm()];
-    s.p = n[i.xp()];
-    s.pp = n[i.xpp()];
+    Stencil1D stencil;
+    stencil.c = n_in[i];
+    stencil.m = n_in[i.xm()];
+    stencil.mm = n_in[i.xmm()];
+    stencil.p = n_in[i.xp()];
+    stencil.pp = n_in[i.xpp()];
 
-    cellboundary(s);
+    cellboundary(stencil);
 
     if ((i.x() == mesh->xend) && (mesh->lastX())) {
       // At right boundary in X
       if (bndry_flux) {
-        BoutReal flux;
-        if (vR > 0.0) {
+        BoutReal flux = BoutNaN;
```

Comment:
performance? Apparently the compiler cannot deduct that flux is always written, and thus cannot optimize the not-needed write ...